File: Python_3.9/1769 - CPF 1.py
# author: Josélio de S. C. Júnior <joseliojrx25@gmail.com>
# copyright: Josélio de S. C. Júnior 2022

import re

def c(l):
  return sum(l) % 11

while True:
  try:
    d1, d2, i, j = [], [], 1, 9
    r = input()
    r = ''.join(re.split('\\.|\\-', r))

    for e in range(9):
      d1.append(int(r[e]) * i)
      d2.append(int(r[e]) * j)
      i += 1
      j -= 1

    d1 = '0' if c(d1) == 10 else str(c(d1))
    d2 = '0' if c(d2) == 10 else str(c(d2))

    if str(r[-2]) + str(r[-1]) == d1 + d2:
      print('CPF valido')
    else:
      print('CPF invalido')

  except EOFError:
    break

File: Pascal_fpc_2.6.2/1769 - CPF 1.pas
// author: Josélio Júnior (Lunatic Fox) <joseliojrx25@gmail.com>
// copyright: Josélio Júnior (Lunatic Fox) 2023

var
  r: string;
  s: string;
  na: array[1..11] of integer;
  i, j, tna, b1, b2: integer;
begin
  while not Eoln do
    begin
    readln(r);
    s := '';
    for i := 1 to 14 do
    begin
      if (r[i] <> '.') and (r[i] <> '-') then
        s := s + r[i];
    end;

    for i := 1 to 11 do
    begin
      tna := ord(s[i]) - 48;
      if (tna >= 0) and (tna <= 9) then
        na[i] := tna;
    end;

    b1 := 0;
    b2 := 0;
    j := 9;
    for i := 1 to 9 do
    begin
      b1 := b1 + (na[i] * i);
      b2 := b2 + (na[i] * j);
      j := j - 1;
    end;
    
    b1 := b1 mod 11;
    b2 := b2 mod 11;
    
    if (b1 = 10) then b1 := 0;
    if (b2 = 10) then b2 := 0;

    if (b1 = na[10]) and (b2 = na[11]) then
      writeln('CPF valido')
    else
      writeln('CPF invalido');
  end;
end.
